src/Hazel/Var.hs

```diff
@@ -0,0 +1,53 @@
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+
+-- | An adaptation of the locally nameless representation (See Charguéraud's
+-- "The Locally Nameless Representation" for more information) which uses 2D
+-- de Bruijn variables.
+
+module Hazel.Var where
+
+import Control.Lens (Iso', Traversal', iso)
+import Data.Word (Word32)
+
+import qualified Data.Text as T
+
+-- | The distance in binding levels between a variable and its binder. The
+-- "first dimension" in our 2D de Bruijn scheme.
+newtype Depth
+  = Depth { _depth :: Word32 }
+  deriving (Eq, Show, Enum)
+
+unDepth :: Iso' Depth Word32
+unDepth = iso (\ (Depth d) -> d) Depth
+{-# INLINE unDepth #-}
```

Comment:
I believe the `INLINE` pragma was copied over and I can't imagine anyone else adding it originally but you :P